app/src/test/java/org/listenbrainz/android/user/UserViewModelTest.kt

```diff
@@ -0,0 +1,49 @@
+package org.listenbrainz.android.user
+
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.test.StandardTestDispatcher
+import kotlinx.coroutines.test.advanceUntilIdle
+import kotlinx.coroutines.test.runTest
+import kotlinx.coroutines.test.setMain
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertNotNull
+import org.junit.Before
+import org.junit.Test
+import org.listenbrainz.android.ui.screens.profile.stats.StatsRange
+import org.listenbrainz.android.ui.screens.profile.stats.UserGlobal
+import org.listenbrainz.android.viewmodel.UserViewModel
+import org.listenbrainz.sharedtest.mocks.MockAppPreferences
+import org.listenbrainz.sharedtest.mocks.MockListensRepository
+import org.listenbrainz.sharedtest.mocks.MockSocialRepository
+import org.listenbrainz.sharedtest.mocks.MockUserRepository
+import org.listenbrainz.sharedtest.utils.EntityTestUtils.testUsername
+
+class UserViewModelTest {
+    private lateinit var viewModel: UserViewModel
+    @OptIn(ExperimentalCoroutinesApi::class)
+    @Before
+    fun setup(){
+        Dispatchers.setMain(StandardTestDispatcher())
+        viewModel = UserViewModel(MockAppPreferences(), MockUserRepository(), MockListensRepository(), MockSocialRepository(), Dispatchers.Default)
+    }
+
+    @OptIn(ExperimentalCoroutinesApi::class)
+    @Test
+    fun getDataTest() = runTest {
+        viewModel.getUserDataFromRemote(testUsername)
+        // Ensure all coroutines and tasks are completed
+        advanceUntilIdle()
+        assertNotNull(viewModel.uiState.value.statsTabUIState)
```

Comment:
Add more comments whenever you add a test so that we can ensure to cover all cases